### ISSUE TITLE
[bug] Building Conan fails with Python 3.12

### ISSUE THREAD

**Issue by gjasny** [conan-io/conan#15114]
### Environment details

* Operating System+version: any
* Compiler+version: any
* Conan version: 2.0.14
* Python version: 3.12

### Steps to reproduce

1. Make sure Python 3.12 is installed
2. Run `pip install -e .`
3. Run `python pyinstaller.py`

With Python 3.12 the deprecated `distutils` package was removed ([PEP 632](https://peps.python.org/pep-0632/#migration-advice)) which now makes the Conan build fail. See: https://github.com/conan-io/conan/blob/889ffbc1243a2542ac87d4761e005e6e0fab8466/pyinstaller.py#L22

### Logs

```
[2023-11-15T21:52:13.937Z] Traceback (most recent call last):
[2023-11-15T21:52:13.937Z]   File "c:\c\ws\6989kc6findlvv8\conan-2.0.14\pyinstaller.py", line 22, in <module>
[2023-11-15T21:52:13.937Z]     from distutils import dir_util
[2023-11-15T21:52:13.937Z] ModuleNotFoundError: No module named 'distutils'
```

**Comment by franramirez688** [comment]
Hi @gjasny 

Thanks for reporting it. Indeed, it was failing because of that. I just removed it from the code because it was not even being used. https://github.com/conan-io/conan/pull/15116 solves it.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/15116 merged, it will be in the next 2.0.15 release, thanks!
